### DISCUSSION TITLE
Open5GS and ZMQ (gNB + UE) setup on separate machines

### DISCUSSION THREAD

**Discussion by ahmadhassandebugs** [srsran/srsRAN_Project#738] [General Help]
Hi all,

I have a setup where Open5GS is running on one machine (M1) and ZMQ-based srsRAN gNB and srsUE on another machine (M2). Both machines are on the local network. I tested the setup and it seems to be working - srsUE is able to establish RRC connection with the core and it creates a tun `tun_srsue` inside the network namespace `ue1`. However, I cannot figure out how to configure routing on M2 so that srsUE can transmit/receive traffic to/from the Open5GS core running on M1. For clarity, here's the summary of the setup:

- M1
  - Open5GS built from source 
- M2
  - srsRAN with ZMQ radio
  - srsUE with ZMQ radio

Note that I was able to successfully run this setup on a single machine after following [ZeroMQ-based Setup](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#zeromq-based-setup) tutorial. 

I think if I can connect `tun_srsue` inside namespace `ue1` to the root (by IP MASQUERADING and FORWARDING), it might solve the issue. Any kind of help/advice will be greatly appreciated.

Thank you!

**Comment by pgawlowicz** [comment]
Hi, does Open5gs listen on correct interface for ngap connection?
Could you share your configs?

**Comment by ahmadhassandebugs** [comment]
Yes, I believe NGAP listens for connections on the correct interface. Here are the configurations (attached in TXT format since github won't let me upload yaml files):

- [sa_base.txt](https://github.com/user-attachments/files/16288751/sa_base.txt)
- [gnb_zmq.txt](https://github.com/user-attachments/files/16288752/gnb_zmq.txt)
- [ue_zmq.txt](https://github.com/user-attachments/files/16288755/ue_zmq.txt)

M1 Open5GS machine has IP address 10.182.50.31 while M2 (gNB and UE) has IP address 10.182.50.201.

Below is a screenshot of M2 terminal when I run the gNB and UE.

![M2 Terminal Screenshot](https://github.com/user-attachments/assets/5b5c12bd-aa92-4513-a29d-ab1ae7966b96)

Thanks!

**Comment by pgawlowicz** [comment]
could you try this to set this parameter:

https://github.com/herlesupreeth/docker_open5gs/blob/8b2f5c9211f37fc9a0d8b1256eec845953a42bb6/upf/upf.yaml#L18C2-L18C38
I think you should set:
```
upf:
...
    gtpu:
      server:
        - address: 127.0.0.4
          advertise: 10.182.50.31
```

**Comment by ahmadhassandebugs** [comment]
I tried it and it's still the same issue. Could you explain why this might have solved the issue?

Please note that I am not using the dockerized open5gs. I built it from scratch and am running the `5gc` binary. 

I think the problem is not because of some Open5GS or srsRAN (mis)configuration, but due to missing routing configuration on M2. This is why I cannot use `tun_srsue` inside `ue1` namespace to ping the root (10.182.50.201). I probably need to find a way to connect `tun_srsue` to the root interface `eno1` (10.182.50.201).

If it is still confusing, I can provide more step-by-step details about the setup and network configuration (`ip a` and `route -n` for root and `ue1` namespace).

**Comment by pgawlowicz** [comment]
ah wait, the UE gets the IP assigned.
Then please try to add a default route to the UE netns:
```
sudo ip netns exec ue1 ip route add default via 10.45.0.1 dev tun_srsue
```

**Comment by ahmadhassandebugs** [comment]
Yes, the IP gets assigned.

I have tried doing that before but it didn't work. Check the screenshot (bottom right tab) below for an overview of routing configurations. Inside `ue1` namespace, I was not able to ping the gateway (10.45.0.1) or the root (10.182.50.201).


![Screenshot from 2024-07-18 14-38-45](https://github.com/user-attachments/assets/e41828d9-01e3-4ad7-9b65-605ce4447be6)

Thanks!

**Reply by pgawlowicz** [reply]
hmm, maybe some firewall rules? or ip forwarding?

**Reply by manasik2019** [reply]
I faced the same issue, it turns out that there is an iptables rule that drops packets from source 10.45.0.0/16, I initially configured it following open5gs guides for building from sources as below: 
![Screenshot from 2024-07-23 10-05-35](https://github.com/user-attachments/assets/b17a3143-32f6-407f-88e0-41f57a259d22)

This solves it: 
``` shell
sudo iptables -L -n -v # list all rules
sudo iptables -D INPUT -s 10.45.0.0/16 -j DROP # remove the one causing issues
```

**Reply by ahmadhassandebugs** [reply]
I never configured these security rules on my setup so that shouldn't be an issue. 

Regardless, I set up the same setup with `docker_open5gs` repository after some struggle so all good :)

**Comment by Gyrusk** [comment]
@ahmadhassandebugs were you able to figure it out?
